pkg/cli/image/manifest/manifest.go

```diff
@@ -40,14 +40,16 @@ func (o *ParallelOptions) Bind(flags *pflag.FlagSet) {
 
 type SecurityOptions struct {
 	RegistryConfig   string
+	PushConfig       string
 	Insecure         bool
 	SkipVerification bool
 
-	CachedContext *registryclient.Context
+	CachedContext []*registryclient.Context
 }
 
 func (o *SecurityOptions) Bind(flags *pflag.FlagSet) {
 	flags.StringVarP(&o.RegistryConfig, "registry-config", "a", o.RegistryConfig, "Path to your registry credentials (defaults to ~/.docker/config.json)")
+	flags.StringVar(&o.PushConfig, "push-config", o.PushConfig, "Path to your registry credentials for pushing images to (defaults to ~/.docker/config.json)")
```

Comment:
nit: Seems like `release new` is more focused on pushing than pulling, so I'd have expected pushing to use `--registry-config` with a new `--pull-config` to override in the pull case (instead of having a `--push-config` to override in the push case).  But obviously is could work either way.

Also, I'd expect (and for backwards compat, we probably need to require) the alternative config to fall back to `--registry-config`, not to fall back straight to `~/.docker/config.json`.  So:

* User sets nothing, we use `~/.docker/config.json` for both.
* User sets `--registry-config`, we use that for both.
* User sets `--push-config`, we use that for push and `~/.docker/config.json` for pull (or vice versa if we go with `--pull-config).
* User sets both, we use the explicitly-provided paths for their respective activities.

Reply:
agreed, i'll update, thanks - also still looking into whether it's possible to use the single auth file _somehow_

Reply:
I've modified this PR: Now, the new flag is `--pull-secret` and if not specified, defaults to `--registry-config` and that defaults to `~/.docker/config.json` 
For users, they should have their Red Hat pull-secret separate from their containers authfile.

Reply:
for example, for a command that only pulls, like `oc adm release extract` you can either
`oc adm release extract -a ~/authfile` or `oc adm release extract --pull-secret ~/authfile`  
(but if you are in habit of pushing releases to a personal quay repo via `release mirror` or `release new`, you _should_ have separate auth files for these as the mirror or release new will require separate auths)